### DISCUSSION TITLE
Change color/add border to floating windows when transparency is on

### DISCUSSION THREAD

**Discussion by apaydev** [LazyVim/LazyVim#4249] [Q&A] [answered]
So, I'm trying to configure transparency using the catppuccin theme. This is my colorscheme.lua file:
```lua
return {
  {
    "catppuccin/nvim",
    name = "catppuccin",
    lazy = "true",
    config = function()
      require("catppuccin").setup({
        flavour = "mocha",
        transparent_background = true,
      })
    end,
  },
  {
    "LazyVim/LazyVim",
    opts = {
      colorscheme = "catppuccin",
    },
  },
}
```

Everything works great, but the floating windows (like hover information and the lazy.nvim panels) are also completely transparent, and that just looks awful.
![image](https://github.com/user-attachments/assets/7a8a468f-7cdd-49b2-ad52-0624b7539fef)

I've tried changing the highlight groups for NormalFloat and FloatBorder, among others, but nothing seems to work. How could I add a white border on those floating windows.

Also, when working with snippets, I would like to change the background color of the right-most panel from black to transparent, and give it a white border. How could I do that?
![image](https://github.com/user-attachments/assets/6fa59e9c-945b-47c0-b627-b9966036fce7)

**Comment by bagashiz** [comment, marked as answer]
you might want to check this out https://github.com/LazyVim/LazyVim/issues/2708

**Reply by bagashiz** [reply]
here is my personal config:

```lua
return {
  {
    "LazyVim/LazyVim",
    opts = {
      colorscheme = "catppuccin",
    },
  },
  {
    "catppuccin",
    opts = {
      transparent_background = true,
    },
  },
  -- border highlight when background is transparent
  {
    "hrsh7th/nvim-cmp",
    opts = function(_, opts)
      opts.window = {
        completion = {
          border = "rounded",
          winhighlight = "Normal:MyHighlight",
          winblend = 0,
        },
        documentation = {
          border = "rounded",
          winhighlight = "Normal:MyHighlight",
          winblend = 0,
        },
      }
    end,
  },
  {
    "williamboman/mason.nvim",
    opts = {
      ui = {
        border = "rounded",
      },
    },
  },
  {
    "folke/noice.nvim",
    opts = {
      presets = {
        lsp_doc_border = true,
      },
    },
  },
  {
    "neovim/nvim-lspconfig",
    opts = {
      diagnostics = {
        float = {
          border = "rounded",
        },
      },
    },
  },
}
```

**Reply by apaydev** [reply]
The `winhighlight` and `winblend` were what I was missing. Thanks a lot!
